server/http/oauth2_complete_test.go

```diff
@@ -0,0 +1,270 @@
+package http
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/jarcoal/httpmock"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/config"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/mocks"
+	"github.com/mattermost/mattermost-plugin-msoffice/server/user"
+	"github.com/mattermost/mattermost-server/app"
+	"github.com/stretchr/testify/assert"
+)
+
+func getHTTPRequest(userID, rawQuery string) *http.Request {
+	r := &http.Request{
+		Header: make(http.Header),
+	}
+
+	r.URL = &url.URL{
+		RawQuery: rawQuery,
+	}
+	r.Header.Add("Mattermost-User-ID", userID)
+
+	return r
+}
+
+func TestOauth2Complete(t *testing.T) {
+	api := &app.PluginAPI{}
+	kv := &mocks.MockKVStore{
```

Comment:
Would you be open, either as a part of this PR, or as a separate one, to instrument all interface mocks to use `github.com/stretchr/testify/mock` like we do elsewhere? I will file a separate issue for it, and explain the history/context there.

Reply:
https://github.com/mattermost/mattermost-plugin-msoffice/issues/3

Reply:
I can do that. I think that it would probably be best to wrap that into this PR.